src/main/java/com/google/sps/models/Chart.java

```diff
@@ -5,11 +5,13 @@
 class Chart {
     private String title;
     private String subtitle;
+    private String id;
     private List<App> apps;
 
     Chart(String title, List<App> apps) {
         this.title = title;
         this.subtitle = subtitle;
+        this.id = id;
         this.apps = apps;
```

Comment:
id and subtitle aren't being properly set. 

if will always set id and subtitle to null. You need to either add a param in the constructor or set up a Builder like in App.java

Reply:
Okay just for clarification set it to null where? I checked the network tab and saw that the information was passed in.

Reply:
Also why would I set a subtitle for each chart if there's not supposed to be one? The subtitle is for the header

Reply:
Lemme rephrase.

When you do `this.subtitle = subtitle;` in the constructor you're actually doing `this.subtitle = null;` because the closest scope for subtitle is the one defined on `private String subtitle;` which is null

We shouldn't even be setting `this.subtitle = subtitle` in the constructor.

We need to pass id into the constructor because of the same scope issue above. To make all this cleaner we should rewrite this using the Builder design pattern like in https://github.com/googleinterns/play-web-step-2020/blob/master/src/main/java/com/google/sps/models/App.java

Reply:
By implementing the builder does that mean to get rid of the constructor and use the Builder function to create the charts?

Reply:
if we go with the Builder route, then you'll have a private Charts constructor that takes in all the args like
```
private Charts(String id, String subtitle, String title, List<App> apps) {
this.id= id;
this.subtitle = subtitle;
this.title = title;
this.apps = apps;
}
```

Other option is to add getters and setters to the Charts like. I prefer the Builder option

```
public Charts(String id) {
this.id = id;
}

public String getTitle() {}
public void setTitle() {}

...

```